### DISCUSSION TITLE
When reading a json field, is the "get<T>()" needed?

### DISCUSSION THREAD

**Discussion by tawmoto** [nlohmann/json#2416] [Q&A] [answered]
Hi,

Reading a json field as string is done like this `auto message = json["details"].get<std::string>();`
But I see that this works also `auto message = json["details"]`, so do I need to put `.get<std::string>()` there?
Same question for other types of course: bool, int, vectors.

Thanks

**Comment by nlohmann** [comment, marked as answer]
`operator[]` returns references of type `json`. So `message` in `auto message = json["details"]` is `json`, but in `auto message = json["details"].get<std::string>();` is `std::string`.

**Reply by tawmoto** [reply]
> `operator[]` returns references of type `json`. So `message` in `auto message = json["details"]` is `json`, but in `auto message = json["details"].get<std::string>();` is `std::string`.

Thank you for your answer.
Because sometimes I have explicitely `std::string message = json["details"]` and it works without any problems, also for `int` is the same.
Is something magical or do you recommend putting everytime the `get<T>()` there?

I like it without `get` because is cleaner, but if you recommend putting the `get`, I will put it, of course.

**Reply by nlohmann** [reply]
If you skip the `get`, but have a type other than `auto`/`json`, then implicit conversions kick in. They are convenient, but can also be the source of bugs, see https://github.com/nlohmann/json#implicit-conversions.
